app/controllers/assessment/autograde.rb

```diff
@@ -71,12 +74,16 @@ def regradeBatch
     submission_ids.each do |submission_id|
       submission = @assessment.submissions.find_by_id(submission_id)
       if submission
-        job = autogradeSubmissions(@course, @assessment, [submission])
-        if job == -2 # no autograding properties for this assessment
-          redirect_to([@course, @assessment, :submissions]) && return
-        elsif job < 0 # autograding failed
-          failed_jobs += 1
-          failed_list += "#{@submission.filename}: autograding error.<br>"
+        begin
+          autogradeSubmissions(@course, @assessment, [submission])
+        rescue AssessmentAutogradeHelper::AutogradeError => e
+          if e.error_code == :missing_autograding_props
+            # no autograding properties for this assessment
+            redirect_to([@course, @assessment, :submissions]) && return
+          else # autograding failed
+            failed_jobs += 1
+            failed_list += "#{@submission.filename}: autograding error.<br>"
```

Comment:
Will this function only be called when rendering HTML? If not, we want to get rid of the `<br>` in case this function is called via API